packages/entities/entities-plugins/src/components/PluginEntityForm.vue

```diff
@@ -0,0 +1,717 @@
+    let modelKey = parent ? `${parent}-${key}` : key
+    let scheme = props.schema[modelKey]
+
+    // If `scheme` is undefined, it is either because the key is not the deepest nested key of the schema object
+    // or the field name has dashes in it and its schema key was converted to underscores during initiation.
+    if (!scheme) {
+      const underscoredModelKey = parent ? `${parent}-${key.replace(/-/g, '_')}` : key.replace(/-/g, '_')
+
+      // Here we check if the underscored key exists in the schema and the `fieldNameHasDashes` flag is true
+      if (props.schema[underscoredModelKey]?.fieldNameHasDashes) {
+        modelKey = underscoredModelKey
+        scheme = props.schema[modelKey]
+      }
+    }
+
+    const value = data[key]
+    const type = typeof value
+
+    // Ensure Object Advanced field is saved to model as empty object
+    if (scheme && scheme.type === 'object-advanced' && !value) {
+      formModel[modelKey] = {}
+      originalModel[modelKey] = {}
+
+      return
+    }
+
+    if (Array.isArray(value)) {
+      formModel[modelKey] = JSON.parse(JSON.stringify(value))
+      originalModel[modelKey] = JSON.parse(JSON.stringify(value))
+
+      return
+    }
+
+    if (type === 'object' && value && !value.length) {
+      return updateModel(value, modelKey)
+    }
+
+    if (type === 'object' && value && value.length && scheme.type === 'input') {
+      formModel[modelKey] = value.join(', ')
+      originalModel[modelKey] = value.join(', ')
+
+      return
+    }
+
+    formModel[modelKey] = value
+    originalModel[modelKey] = value
+  })
+
+  emit('model-updated', {
+    model: formModel,
+    originalModel,
+    data: getModel(),
+  })
+}
+
+const getModel = (): Record<string, any> => {
+  const schema = { ...props.schema }
+  const inputModel = formModel
+  const origModel = originalModel
+  const formModelFields = Object.keys(inputModel)
+  const outputModel: Record<string, any> = {}
+
+  if (!isObjectEmpty(formSchema.value)) {
+    // convert formSchema from array to object
+    for (let i = 0; i < formSchema.value?.fields?.length; i++) {
+      schema[formSchema.value.fields[i].model] = formSchema.value.fields[i]
+    }
+  }
+
+  // Moved custom field keys to top level of schema & remove key
+  customFields.forEach((customField: string) => {
+    if (schema[customField]) {
+      schema[customField].fields.forEach((field: Record<string, any>) => {
+        // if the field has a fields array, set each to top level.
+        // Check out buildCustomFields() in EntityMixin for similar logic
+        if (field.fields) {
+          field.fields.forEach((subField: Record<string, any>) => {
+            schema[subField.model] = subField
+          })
+        }
+
+        if (field.model) schema[field.model] = field
+      })
+      delete schema[customField]
+    }
+  })
+
+  // Iterate over each field, transform each field value to match the
+  // Kong Admin APIs value expectations
```

Comment:
question: how do we know the logic below matches what the Admin API is expecting?

Reply:
Because we make it? This is the logic to transform the formdata into the request structure expected for the API